tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
+      jqElement =
+        selector: selector
+        data: sinon.spy()
+        scrollintoview: sinon.spy (options) ->
+          scrollTarget = this.selector
+          options?.complete?()
+
+  afterEach ->
+    sandbox.restore()
+    scrollTarget = null
+
+  describe "constructor", ->
+    it 'wraps a highlight span around the given range', ->
+      hl = createTestHighlight()
+      assert.calledWith Annotator.TextHighlight.highlightRange, "test range"
+
+    it 'stores the created highlight spans in _highlights', ->
```

Comment:
> This still points in the direction of figuring out how to expose `Element`, or a collection thereof 

I think I actually had that, before someone "simplified" the abstract Highlight class and the TextHighlight class into one shortened version. See [here](https://github.com/hypothesis/h/pull/1868/files#diff-130344517827f6cc46967dbcc766d7b1L215).

> and `scrollIntoView` not being on the highlight class.

I am not sure that's s good idea, because ultimately, there is no guarantee that we are talking about DOM elements at all. For example, we could be talking about rectangles being drawn to a canvas, over a given region of an image... that's why I think that ultimately, it's up to the concrete Highlight implementation to implement scrolling. 

> For instance, we could create highlight elements for virtual highlights and patch their implementation to do things like scroll to the page, then the highlight, as you have done elsewhere.

We could, but what would be the benefit of this, compared to simply using the Anchor's `scrollIntoView` method? The Anchor is simply a higher level abstraction of the Highlight. You can always ought to the Anchor's `scrollIntoView()` method, and it will take care of everything.